.github/workflows/publish.yml

```
@@ -1,43 +1,35 @@
-name: Publish
+name: Test and Release Go CLI
 
 on:
   push:
 
 jobs:
-  publish:
+  build:
+    name: GoReleaser build
     runs-on: ubuntu-latest
-    env:
-      GOX_ARCHITECTURES: amd64 386
-    steps:
-      # Currently there is no possibility to get a tag name. Hence we filter the GITHUB_REF variable which contains something like "refs/tags/v2.0.0" if the tag is "v2.0.0". The value can be used with "steps.tag_name.outputs.TAG_NAME"
-      - name: Get tag name
-        if: startsWith(github.ref, 'refs/tags/')
-        id: tag_name
-        run: |
-          echo "::set-output name=TAG_NAME::${GITHUB_REF#refs/tags/}"
-          echo "$TAG_NAME"
 
-      - name: Checkout the code
+    steps:
+      - name: Check out code into the Go module directory
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # See: https://goreleaser.com/ci/actions/
 
       - name: Setup Go environment
         uses: actions/setup-go@v3
         with:
           go-version-file: "go.mod"
           check-latest: true
+        id: go
       - run: go version
 
       - name: Execute the tests
         run: go test -race ./...
 
-      - name: Build the binaries
-        if: startsWith(github.ref, 'refs/tags/')
-        run: |
-          go install github.com/mitchellh/gox@latest
-          ./dists/make-release.sh -v ${{ steps.tag_name.outputs.TAG_NAME }}
-
-      - name: Release the new version
+      - name: Run GoReleaser
         if: startsWith(github.ref, 'refs/tags/')
-        uses: softprops/action-gh-release@v1
+        uses: goreleaser/goreleaser-action@master
```

I would prefer you use a version number here rather than `master`

```suggestion
        uses: goreleaser/goreleaser-action@v3
```